src/transformers/models/owlv2/convert_owlv2_to_hf.py

```diff
@@ -0,0 +1,331 @@
+    model.eval()
+
+    # for name, param in model.named_parameters():
+    #     print(name, param.shape)
+
+    # Initialize image processor
+    size = {"height": config.vision_config.image_size, "width": config.vision_config.image_size}
+    image_processor = Owlv2ImageProcessor(size=size)
+    # Initialize tokenizer
+    tokenizer = CLIPTokenizer.from_pretrained("openai/clip-vit-base-patch32", pad_token="!", model_max_length=16)
+
+    # Initialize processor
+    processor = Owlv2Processor(image_processor=image_processor, tokenizer=tokenizer)
+
+    filename = "owlvit_pixel_values_960.pt" if "v2" in model_name else "owlvit_pixel_values.pt"
+    filepath = hf_hub_download(repo_id="nielsr/test-image", filename=filename, repo_type="dataset")
+    pixel_values = torch.load(filepath).permute(0, 3, 1, 2)
+    print("Shape of pixel values:", pixel_values.shape)
+    filename = "owlv2_input_ids.pt" if "v2" in model_name else "owlvit_input_ids.pt"
+    filepath = hf_hub_download(repo_id="nielsr/test-image", filename=filename, repo_type="dataset")
+    input_ids = torch.load(filepath).squeeze()
+    print("Shape of input ids:", input_ids.shape)
+
+    with torch.no_grad():
+        outputs = model(input_ids=input_ids, pixel_values=pixel_values)
+        logits = outputs.logits
+        pred_boxes = outputs.pred_boxes
+
+    # # note: the logits below were obtained without center cropping
+    # if checkpoint_url == "https://dl.fbaipublicfiles.com/convnext/owlvit/im1k/owlvit_atto_1k_224_ema.pt":
+    #     expected_logits = torch.tensor([-0.3930, 0.1747, -0.5246, 0.4177, 0.4295])
+    # else:
+    #     raise ValueError(f"Unknown URL: {checkpoint_url}")
+
+    if model_name == "owlvit-base-patch16":
+        expected_logits = torch.tensor(
+            [[-9.6403, -9.6249, -9.1869], [-12.4517, -11.8114, -12.3912], [-12.2289, -11.3871, -12.0908]]
+        )
+        expected_boxes = torch.tensor([[0.0206, 0.0355, 0.0400], [0.0711, 0.3572, 0.1431], [0.0811, 0.4575, 0.1718]])
+    elif model_name == "owlv2-base-patch16":
+        expected_logits = torch.tensor(
+            [[-10.0043, -9.0226, -8.0433], [-12.4569, -14.0380, -12.6153], [-21.0731, -22.2705, -21.8850]]
+        )
+        expected_boxes = torch.tensor([[0.0136, 0.0223, 0.0269], [0.0406, 0.0327, 0.0797], [0.0638, 0.1539, 0.1255]])
+    elif model_name == "owlv2-base-patch16-ensemble":
+        expected_logits = torch.tensor(
+            [[-8.6353, -9.5409, -6.6154], [-7.9442, -9.6151, -6.7117], [-12.4593, -15.3332, -12.1048]]
+        )
+        expected_boxes = torch.tensor([[0.0126, 0.0090, 0.0238], [0.0387, 0.0227, 0.0754], [0.0582, 0.1058, 0.1139]])
+
+    print("Logits:", logits[0, :3, :3])
+    print("Pred boxes:", pred_boxes[0, :3, :3])
+
+    assert torch.allclose(logits[0, :3, :3], expected_logits, atol=1e-3)
+    assert torch.allclose(pred_boxes[0, :3, :3], expected_boxes, atol=1e-3)
+    print("Looks ok!")
```

Comment:
Thanks for updating! 
I still think we should allow to disable this test with a kwarg, but if indeed the framework used fo owl is not used a lot / won't produce other checkpoints then no worries!